src/contact.rs

```diff
@@ -699,15 +696,13 @@ impl Contact {
                 .query_map(
                     "SELECT c.id FROM contacts c \
                  LEFT JOIN acpeerstates ps ON c.addr=ps.addr  \
-                 WHERE c.addr!=?1 \
```

Comment:
is it safe to remove the check. there is one corner case maybe:  you have a contact whose address later becomes your primary or secondary address.  Maybe this means we need to check/modify the contacts table when setting a new primary address, to check if the primary address is by chance the addr of an existing contact?

Reply:
This corner case is a problem, yes.

> Maybe this means we need to check/modify the contacts table when setting a new primary address, to check if the primary address is by chance the addr of an existing contact?

+1

Reply:
what would we do though? Locally remove the offending contact (now primary) from all chat groups?  We said that we do not want to cause network messages with the re-configure so we can not send out a remove-member message out. 
how about: 
- On the next message with a aeap-replacement header the receiving side could take care of removing the contact from the respective chat group where the message arrived. 
- such outgoing messages with replacement header also cause the local group to remove the primary contact as a member which is a duplicate of the configured primary address.

Reply:
> such outgoing messages with replacement header also cause the local group to remove the primary contact as a member which is a duplicate of the configured primary address.

But we said above that when setting a new primary addr, we already have to delete the corresponding "primary" contact from the db during configure to make the sql query work fine. How about:
- When adding a self address, the corresponding contact is removed locally from all groups _(or replaced with `ContactId::SELF` if `ContactId::SELF` was not a member)_ and the contact is deleted. Idk what to do with messages in the 1:1 chat with this contact - create a group for them?. _This means that the member state is temporarily inconsistent, but that's OK since it will be restored at the next outgoing message:_
- On the next message with an aeap-replacement header the receiving side takes care of removing the now-secondary contact from the respective group where the message arrived.

Reply:
i'd like to avoid adding new flags and too many membership-shenenigans.  How i currently suggest things working overall: 

1. (mostly DONE) At the time of configure we push the current primary to become a secondary. The sql-query can be made to work like before (if even needed, i am not sure)

2. When a message is sent out to a chat, and the message is encrypted, and we have secondary addresses, then we 
  a) add a protected "AEAP-Replacement" header that contains all secondary addresses 
  b) if any of the secondary addresses is in the chat's member list, we remove it and leave a system message that we did so
3. When an encrypted message with a replacement header is received, replace the e-mail address of all secondary contacts (if they exist) with the new primary and drop a sysmessage in all chats the secondary is member off.  This might (in edge cases) result in chats that have two or more contacts with the same e-mail address.  We might ignore this for a first release and just log a warning.  Let's maybe not get hung up on this case before everything else works. 

Notes: 
- for now we will send out aeap replacement headers forever, there is no termination condition other than lack of secondary addresses.  I think that's fine for now.  Later on we might introduce options to remove secondary addresses but i wouldn't do this for a first release/PR. 
- the design is resilient against changing e-mail providers from A to B to C and then back to A, with partially updated chats and diverging views from recipients/contacts on this transition.  In the end, you will have a primary and some secondaries, and when you start sending out messages everybody will eventually synchronize when they receive the current state of primaries/secondaries. 
- there are no extra flags/columns in the database needed (i hope)

Reply:
What we discussed today:

We do it as @hpk42 proposes here. This means that we accept these disadvantages:
- Inconsistent group state: Suppose Alice does an AEAP transition and sends a 1:1 message to Bob, so Bob rewrites Alice's contact. Alice, Bob and Charlie are together in a group. Before Alice writes to this group, Bob and Charlie will have different membership lists, and Bob will send messages to Alice's new address, while Charlie will send them to her old address.
- There will be multiple contacts with the same address in the database.

The most obvious alternative would be to create a new contact with the new address and replace the old contact in the groups.

We accept the disadvantages for these advantages:
- From there, it's easier to switch to a model where the info about the transition is encoded in the PGP key. Since the key is gossiped, the information about the transition will spread virally.
- (Also, less important: Slightly faster transation: If you send a message to e.g. "Delta Chat Dev", all members of the "sub-group" "delta android" will know of your transition.)
- It's easier to implement (if too many problems turn up, we can still switch to another approach.